hera_sim/simulate.py

```diff
@@ -283,24 +279,29 @@ def add_foregrounds(self, model, **kwargs):
 
         Args:
             model (str or callable): either a string name of a model function existing in :mod:`~hera_sim.foregrounds`,
-                or a callable which has the signature ``fnc(lsts, fqs, bl_len_ns, **kwargs)``.
+                or a callable which has the signature ``fnc(lsts, fqs, bl_vec, **kwargs)``.
             ret_vis (bool, optional): whether to return the visibilities that are being added to to the base
                 data as a new array. Default False.
             add_vis (bool, optional): whether to add the calculated visibilities to the underlying data array.
                 Default True.
-            **kwargs: keyword arguments sent to the foregournd model function, other than `lsts`, `fqs` and `bl_len_ns`.
+            **kwargs: keyword arguments sent to the foregournd model function, other than `lsts`, `fqs` and `bl_vec`.
         """
+        # frequencies come from zeroths spectral window
+        fqs = self.data.freq_array[0] * 1e-9
+
+        # prep kwargs
+        if model.__name__ == 'diffuse_foreground':
```

Comment:
this wouldn't be necessary if either you 1) use keyword args when passing to the model function, or 2) specify the ``diffuse_foregrounds`` function with ``Tsky_mdl`` not as the first argument.

Reply:
I think this is redundant with the comment you made above. I still don't see what you mean because 1) I am passing keyword args to the model function.

Do you mean that this would work if I define `Tsky_mdl` to be a kwarg in the original function `diffuse_foreground()`? That I think would work, but b/c `Tsky_mdl` is a required parameter for that function, it seems counterintuitive to not make it an argument

Reply:
Yeah, this is redundant with the previous comment. I agree that there is a bit of ambiguity here as to the best way to do it (obviously your way works just as well as the one I've mentioned above). Having thought a little bit more about it, I think the reason I like the way I described above slightly more is because of encapsulation -- the API function doesn't contain information about the specific model its calling -- all of the information is contained within the model function itself. The model itself raises the error if it hasn't been called correctly. This means we can write arbitrarily many foreground model functions and not have to change the ``Simulator`` class. It also means that users can write their own foreground model functions and plug-n-play the same way without touching the main code. 

On the other hand, I agree that making ``Tsky_mdl`` a keyword parameter is a bit counter-intuitive. It could be made a required argument even in the case I outlined above, of course, at the expense of less-informative errors when it isn't passed correctly.